wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/jpa/JpaFactoryDao.java

```diff
@@ -0,0 +1,155 @@
+            if (factory == null) {
+                throw new NotFoundException(format("Factory with id '%s' doesn't exist", id));
+            }
+            return factory;
+        } catch (RuntimeException ex) {
+            throw new ServerException(ex.getLocalizedMessage(), ex);
+        }
+    }
+
+    @Override
+    @Transactional
+    public List<FactoryImpl> getByAttribute(int maxItems,
+                                            int skipCount,
+                                            List<Pair<String, String>> attributes) throws ServerException {
+        try {
+            LOG.info("FactoryDao#getByAttributes #maxItems: {} #skipCount: {}, #attributes: {}", maxItems, skipCount, attributes);
+            final Map<String, Pair<String, String>> params = new HashMap<>();
+            for (Pair<String, String> attribute : attributes) {
+                params.put(NameGenerator.generate("att", 3), attribute);
+            }
+            String query = "Select factory From Factory factory%1$2s %2$2s";
+            if (!params.isEmpty()) {
+                final StringJoiner from = new StringJoiner(", ", ", ", "");
+                final StringJoiner where = new StringJoiner(" AND ", "Where ", " ");
+                for (Map.Entry<String, Pair<String, String>> entry : params.entrySet()) {
+                    from.add(format("factory.%1$2s %2$2s", entry.getValue().first, entry.getKey()));
+                    where.add(entry.getKey() + " = '" + entry.getValue().second + "'");
+                }
+                query = format(query, from, where);
+            } else {
+                query = format(query, "", "");
+            }
+            return managerProvider.get()
+                                  .createQuery(query, FactoryImpl.class)
+                                  .setFirstResult(skipCount)
+                                  .setMaxResults(maxItems)
+                                  .getResultList();
+        } catch (RuntimeException ex) {
+            throw new ServerException(ex.getLocalizedMessage(), ex);
+        }
+    }
+
+    @Transactional
+    protected FactoryImpl doCreate(FactoryImpl factory) {
+        final EntityManager manager = managerProvider.get();
+        manager.persist(factory);
+        return manager.find(FactoryImpl.class, factory.getId());
+    }
+
+    @Transactional
+    protected FactoryImpl doUpdate(FactoryImpl update) throws NotFoundException {
+        final EntityManager manager = managerProvider.get();
+        if (manager.find(FactoryImpl.class, update.getId()) == null) {
+            throw new NotFoundException(format("Could not update factory with id %s because it doesn't exist", update.getId()));
+        }
+        return manager.merge(update);
+    }
+
+    @Transactional
+    protected void doRemove(String id) {
+        final EntityManager manager = managerProvider.get();
+        final FactoryImpl factory = manager.find(FactoryImpl.class, id);
+        if (factory != null) {
```

Comment:
fix doc or add throwing of NotFoundException https://github.com/eclipse/che/pull/1926/files#diff-7c26382a725a0abec59646c16536c728R69